arelle/XbrlUtil.py

```diff
@@ -49,14 +63,16 @@ def equalityHash(elt, equalMode=S_EQUAL, excludeIDs=NO_IDs_EXCLUDED):
                 return elt._hashXpathEqual
         except AttributeError:
             dts = elt.modelXbrl
+            from arelle.ModelXbrl import ModelXbrl
+            assert isinstance(dts, ModelXbrl), 'dts is not an instance of ModelXbrl'
             if not hasattr(elt,"xValid"):
-                xmlValidate(dts, elt)
-            hashableValue = elt.sValue if equalMode == S_EQUAL else elt.xValue
+                xmlValidate(dts, elt)  # type: ignore[no-untyped-call]
+            hashableValue = getattr(elt, 'sValue') if equalMode == S_EQUAL else getattr(elt, 'xValue')
```

Comment:
Why the change to `getattr`?

Reply:
That was before adding the `xValue` and `sValue` attrs to `ModelObject`, missed reverting this one